source/_posts/2017/2017-05-05-how-to-refactor-to-new-dependency-injection-features-in-symfony-3-3.md

```diff
@@ -0,0 +1,251 @@
+
+    some_controller:
+        class: App\Controller\SomeController
+    
+    first_repository:
+        class: App\Repository\FirstRepository
+        calls:
+            - ["setEntityManager", ["@entity_manager"]]
+    second_repository:
+        class: App\Repository\FirstRepository
+        calls:
+            - ["setEntityManager", ["@entity_manager"]]
+         
+    # console commands
+    first_command:
+        class: App\Command\FirstCommand
+        tags:
+            - { name: console.command }
+    second_command:
+        class: App\Command\SecondCommand
+        tags:
+            - { name: console.command }
+         
+    # event subscribers
+    first_subscriber:
+        class: App\EventSubscriber\FirstSubscriber
+        tags:
+            - { name: kernel.event_subscriber }
+    second_command:
+        class: App\EventSubscriber\SecondSubscriber
+        tags:
+            - { name: kernel.event_subscriber }
+```
+
+### 2. Use autoconfigure
+
+```yaml
+# app/config/services.yml
+services:
+    _defaults:
+        autowired: true
+        autoconfigure: true # all Symfony native tags are now added automatically  
+        
+    some_service:
+        class: App\SomeService
+
+    some_controller:
+        class: App\Controller\SomeController
+    
+    first_repository:
+        class: App\Repository\FirstRepository
+        calls:
+            - ["setEntityManager", ["@entity_manager"]]
+    second_repository:
+        class: App\Repository\FirstRepository
+        calls:
+            - ["setEntityManager", ["@entity_manager"]]
+         
+    # console commands
+    first_command:
+        class: App\Command\FirstCommand
+    second_command:
+        class: App\Command\SecondCommand
+         
+    # event subscribers
+    first_subscriber:
+        class: App\EventSubscriber\FirstSubscriber
+    second_command:
+        class: App\EventSubscriber\SecondSubscriber
+```
+
+### 3. Use Class-Named Services
+
+```yaml
+# app/config/services.yml
+services:
+    _defaults:
+        autowired: true
+        autoconfigure: true
+        
+    App\SomeService: ~ # no more thinking about creative and unique service name
+
+    App\Controller\SomeController: ~
+    
+    App\Repository\FirstRepository: 
+        calls:
+            - ["setEntityManager", ["@entity_manager"]]
+    App\Repository\FirstRepository: 
+        calls:
+            - ["setEntityManager", ["@entity_manager"]]
+         
+    # console commands
+    App\Command\FirstCommand: ~
+    App\Command\SecondCommand: ~
+         
+    # event subscribers
+    App\EventSubscriber\FirstSubscriber: ~
+    App\EventSubscriber\SecondSubscriber: ~
+```
+
+### 4. Use PSR-4 based service autodiscovery and registration
+
+```yaml
+# app/config/services.yml
+services:
+    _defaults:
+        autowired: true
+        autoconfigure: true
+        
+    App\: # no more manual registration of similar groups of services 
+        resource: ../{ Controller, Command, Subscriber }
```

Comment:
I got some inner latte parsing issues (that are difficult to debug and not priority at the moment), so I put spaces there.

Will try without with sign hack then.